frontend/src/pages/_app.tsx

```diff
@@ -135,10 +135,16 @@ const MenuContent: React.FC<{ hasAdminPermission: boolean }> = ({ hasAdminPermis
 const RoutingMiddleware: React.FC<{ children: ReactNode }> = ({ children }) => {
   const router = useRouter();
   const hasMemberPermissions = useHasMemberPermission();
+  const hasAdminPermission = useHasAdminPermission();
 
-  if (!router.pathname.startsWith('/applicant') && !hasMemberPermissions) {
-    router.push('/applicant');
-  }
-
+  useEffect(() => {
+    if (!router.pathname.startsWith('/applicant') && !hasMemberPermissions) {
+      router.push('/applicant');
```

Comment:
```suggestion
    if (router.pathname !== '/applicant') router.push('/applicant');
```
This is just to check if there already on the page to prevent the push.